[Breaking] Daikin176: Change operating mode values.

crankyoldgit opened the pull request:
* More user supplied data has shown the `altmode` was the real mode all along. Change code to reflect that.
* Change mode values to use `kDaikin176Fan` etc.
* Add support for Heat & Auto modes.
* Update unit tests accordingly.
* Update the supported model info.

Fixes #1233

crankyoldgit committed:
[Breaking] Daikin176: Change operating mode values
* More data has shown the `altmode` was the real mode all along. Change code to reflect that.
* Change mode values to `kDaikin176Fan` etc.
* Add support for Heat & Auto modes.
* Update unit tests accordingly.

Fixes #1233

crankyoldgit committed:
Update the daikin supported model info.

NiKiZe commented on src/ir_Daikin.cpp:
```diff
@@ -2322,34 +2322,38 @@ void IRDaikin176::off(void) { setPower(false); }
-    case kDaikinDry: altmode = 7; break;
-    case kDaikin176Cool: altmode = 2; break;
-    default: this->setMode(kDaikin176Cool); return;
+    case kDaikin176Dry:  altmode = 2; break;
```
Never noticed this altmode before, quite magic without comments?

crankyoldgit replied:
Fair call. Added some comments to document it.

crankyoldgit committed:
Code review feedback.
* Additional code comments to document the alternative mode bits.

crankyoldgit committed:
Daikin176: Adjust `.toString()` output
For https://github.com/crankyoldgit/IRremoteESP8266/issues/1233#issuecomment-671009026

crankyoldgit committed:
Confirmed working.
https://github.com/crankyoldgit/IRremoteESP8266/issues/1233#issuecomment-671021199

crankyoldgit commented:
Confirmed working in https://github.com/crankyoldgit/IRremoteESP8266/issues/1233#issuecomment-671021199
